lib/class-wp-theme-json-gutenberg.php

```diff
@@ -3869,4 +3869,41 @@ function ( $carry, $item ) {
 		$theme_json->theme_json['styles'] = self::convert_variables_to_value( $styles, $vars );
 		return $theme_json;
 	}
+
+	/**
+	 * Generates a selector for a block style variation.
+	 *
+	 * @param string $variation_name Name of the block style variation.
+	 * @param string $block_selector CSS selector for the block.
+	 *
+	 * @return string Block selector with block style variation selector added to it.
+	 */
+	public static function get_block_style_variation_selector( $variation_name, $block_selector ) {
+		$variation_class = ".is-style-$variation_name";
+
+		if ( ! $block_selector ) {
+			return $variation_class;
+		}
+
+		$selector_parts = explode( ',', $block_selector );
+		$result         = array();
+
+		foreach ( $selector_parts as $part ) {
+			$class_added = false;
+			$result[]    = preg_replace_callback(
+				'/((?::is|not|has|where\([^)]+\))?\s*)([^\s:]+)/',
+				function ( $matches ) use ( $variation_class, &$class_added ) {
+					if ( $class_added ) {
+						return $matches[0];
+					}
+
+					$class_added = true;
```

Comment:
Does this always result in replacement happening in the first round of calling `preg_replace_callback`? Just wondering if this is much different to passing a `$limit` param of `1` to `preg_replace_callback` 🤔

Apologies if I'm missing nuance here!

Reply:
No, it was me missing things sorry. Just how it came out after a few failed attempts. 

Switching this to use `limit` cleans things up nicely.

Reply:
Nice. For what it's worth, I didn't know about the `limit` param until I started reviewing this and wondered how the `preg_replace_callback` function works and found it in the docs 😄